### PR TITLE
chore: clean up godoc and test coverage output

### PR DESCRIPTION
**Pull request by morningvera**
Added .gitignore because of test coverage output and moved godoc package description to a separate file (doc.go)

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// chain, and eris.Cause returns the root cause of the error. Currently,
+// eris.Is works simply by comparing error messages with each other. If an
+// error contains a particular error message anywhere in its chain (e.g. "not
+// found"), it's defined to be that error type (i.e. eris.Is will return
+// true).
+//
+//    NotFound := eris.New("not found")
+//    _, err := db.Get(id)
+//    // check if the resource was not found
+//    if eris.Is(err, NotFound) || eris.Cause(err) == NotFound {
```
Comment (sum2000) [inline doc.go]: separate the Is and Cause in different lines or methods

File: doc.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+//    if err != nil {
+//      return eris.Wrapf(err, "error getting resource '%v'", id)
+//    }
+//    fmt.Printf("%v", err) // print without the stack trace
+//    fmt.Printf("%+v", err) // print with the stack trace
+//
+// For an error that has been wrapped once, the output will look something
+// like this:
+//
+//    # output without the stack trace
```
Comment (sum2000) [inline doc.go]: Replace with new stack trace

File: doc.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+//
+//    var fields log.Fields
+//    unpackedErr := eris.Unpack(err)
+//    fields["method"] = "api.GetResource"
+//    fields["error"] = unpackedErr.ToJSON(eris.NewDefaultFormat(true))
+//    logger.WithFields(fields).Errorf("method completed with error (%v)", err)
+//
+// When using a JSON logger, the output should look something like this:
+//
+//    {
```
Comment (sum2000) [inline doc.go]: Replace this block with new format

**Comment by sum2000** [comment]
Readme also needs to be updated with latest stack. I suggest we merge the other PR first and then update the README and doc.go otherwise we need to revisit this again

File: doc.go (hunk 2 above)
Comment (morningvera) [reply doc.go]: there is no new stack trace yet

File: doc.go (hunk 1 above)
Comment (morningvera) [reply doc.go]: done

**Comment by morningvera** [comment]
closes issue #39

**Pull request merged**